{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyNxAcXOFp+mfIDl2AtyJ+NO",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/amien1410/amien-scrapers/blob/main/Yelp_Scraper_Okt_2025.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 35
        },
        "id": "5ec03d26",
        "outputId": "8f862861-c24b-4728-b192-7819562b194b"
      },
      "source": [
        "import requests\n",
        "import json\n",
        "\n",
        "# Replace with your GraphQL endpoint URL\n",
        "graphql_url = \"https://www.yelp.com/gql/batch\"\n",
        "\n",
        "# Prepare the request payload\n",
        "payload = [\n",
        "  {\n",
        "    \"operationName\": \"GetResultsSectionData\",\n",
        "    \"variables\": {\n",
        "      \"BizEncIds\": [\n",
        "        \"HdZLo3v4zNVpzx1CQjg5MA\",\n",
        "        \"2rvzTYaiXlqtBe5ZhNDiXA\",\n",
        "        \"iUockw0CUssKZLyoGJYEXA\",\n",
        "        \"QueFVMcMlT-6aZFv2M47mg\",\n",
        "        \"f-m7-hyFzkf0HSEeQ2s-9A\",\n",
        "        \"THpTY0X_xwRzUL7wYD6dDg\",\n",
        "        \"JAHzjTlZceEypYTnGziIjw\",\n",
        "        \"8dUaybEPHsZMgr1iKgqgMQ\",\n",
        "        \"mSMZJj2pFvttWLpcDmgrEA\",\n",
        "        \"uUifmxOmL8ClHXyRVsmD3Q\",\n",
        "        \"0T1Kc6tUWaG3cPnyiD40Hw\",\n",
        "        \"B09WOy0W83Od-Xw4xEXxog\",\n",
        "        \"ZSzXw0NgJTyOzcHwKY5eMA\",\n",
        "        \"yPxYIAJXOJJtpD1YT5fSIA\"\n",
        "      ]\n",
        "    },\n",
        "    \"extensions\": {\n",
        "      \"operationType\": \"query\",\n",
        "      \"documentId\": \"504b9952f2e7a3d797f46552fe260445cc9b0078bcacbfea5dc578be4c42cd41\"\n",
        "    }\n",
        "  }\n",
        "]\n",
        "\n",
        "# Make the POST request\n",
        "response = requests.post(graphql_url, json=payload)\n",
        "\n",
        "# Check for successful response\n",
        "# if response.status_code == 200:\n",
        "#     data = response.json()\n",
        "#     print(\"Response data:\")\n",
        "#     print(json.dumps(data, indent=2))\n",
        "# else:\n",
        "#     print(f\"Error: {response.status_code}\")\n",
        "#     print(response.text)\n",
        "\n",
        "data = response.json()[0]['data']['businesses']\n",
        "data[0]['name']"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Capital One Caf√©'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "\n",
        "url = \"https://www.yelp.com/search?find_desc=restaurant&find_loc=San+Francisco%2C+CA&dd_referrer=https%3A%2F%2Fwww.yelp.com%2F&start=0\"\n",
        "headers = {\n",
        "  \"accept\": \"text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7\",\n",
        "  \"accept-encoding\": \"gzip, deflate, br, zstd\",\n",
        "  \"accept-language\": \"en-US,en;q=0.9,id;q=0.8,fa;q=0.7,ar;q=0.6,ms;q=0.5,ja;q=0.4,es;q=0.3\",\n",
        "  \"cookie\": \"wdi=2|52ABFD0420946339|0x1.a3cc291b0966cp+30|f13f0ac0eca372ef; hl=en_US; bse=bfdc544271584567b1e9404995a76c24; recentlocations=San+Francisco%2C+CA; location=%7B%22county%22%3A+%22San+Francisco+County%22%2C+%22max_longitude%22%3A+-122.3550796508789%2C+%22location_type%22%3A+%22locality%22%2C+%22max_latitude%22%3A+37.81602226140252%2C+%22latitude%22%3A+37.775123%2C+%22min_longitude%22%3A+-122.51781463623047%2C+%22address1%22%3A+%22%22%2C+%22city%22%3A+%22San+Francisco%22%2C+%22display%22%3A+%22San+Francisco%2C+CA%22%2C+%22parent_id%22%3A+371%2C+%22place_id%22%3A+%221237%22%2C+%22state%22%3A+%22CA%22%2C+%22country%22%3A+%22US%22%2C+%22unformatted%22%3A+%22San+Francisco%2C+CA%22%2C+%22address2%22%3A+%22%22%2C+%22longitude%22%3A+-122.41932%2C+%22provenance%22%3A+%22YELP_GEOCODING_ENGINE%22%2C+%22min_latitude%22%3A+37.706368356809776%2C+%22accuracy%22%3A+4%2C+%22zip%22%3A+%22%22%2C+%22address3%22%3A+%22%22%2C+%22borough%22%3A+%22%22%2C+%22confident%22%3A+null%2C+%22isGoogleHood%22%3A+false%2C+%22language%22%3A+null%2C+%22neighborhood%22%3A+%22%22%2C+%22polygons%22%3A+null%2C+%22usingDefaultZip%22%3A+false%7D; spses.5f33=*; bsi=1%7C7fe57c3f-c591-5cf9-b41c-17371e2b0b0a%7C1760834416237%7C1760833851870%7C1%7C276ad153e492ce38; datadome=Mm4N0I12bStLHERGojC~73naXU4qvNtFuUXTP6W~hHgVplNti8j_Wa_MqqL2afVdXvdwjcfQ3CWRHivKgWzEHO04n5m7Wq~maVmjoO0FAnnzUEIUQuXFKHmG~uN0U5v_; xcj=1|-v1GExBZhZkowwMA3liee3GDGp9DTaO4Bs113OHxIcM; g_state={\\\"i_l\\\":0,\\\"i_ll\\\":1760834439076,\\\"i_b\\\":\\\"v0Xf9ckkhoEKjefyz0tgf0VkqzAK06le2G8l/jcqQWo\\\"}; spid.5f33=efd77de1-17f1-4cdb-84d5-f403172632bc.1760758353.2.1760834522.1760759976.9a40d405-7e86-42f6-b457-162b28ffead3.90a65d37-2fa4-4b6a-9422-c3f59bbb6e31.6d056d19-f6af-4bc3-8b23-7cbe2b05d89b.1760833871587.61\",\n",
        "  \"priority\": \"u=0, i\",\n",
        "  \"sec-ch-device-memory\": \"4\",\n",
        "  \"sec-ch-ua\": \"\\\"Microsoft Edge\\\";v=\\\"141\\\", \\\"Not?A_Brand\\\";v=\\\"8\\\", \\\"Chromium\\\";v=\\\"141\\\"\",\n",
        "  \"sec-ch-ua-arch\": \"\\\"x86\\\"\",\n",
        "  \"sec-ch-ua-full-version-list\": \"\\\"Microsoft Edge\\\";v=\\\"141.0.3537.71\\\", \\\"Not?A_Brand\\\";v=\\\"8.0.0.0\\\", \\\"Chromium\\\";v=\\\"141.0.7390.66\\\"\",\n",
        "  \"sec-ch-ua-mobile\": \"?0\",\n",
        "  \"sec-ch-ua-model\": \"\\\"\\\"\",\n",
        "  \"sec-ch-ua-platform\": \"\\\"Windows\\\"\",\n",
        "  \"sec-fetch-dest\": \"document\",\n",
        "  \"sec-fetch-mode\": \"navigate\",\n",
        "  \"sec-fetch-site\": \"none\",\n",
        "  \"sec-fetch-user\": \"?1\",\n",
        "  \"upgrade-insecure-requests\": \"1\",\n",
        "  \"user-agent\": \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/141.0.0.0 Safari/537.36 Edg/141.0.0.0\"\n",
        "}\n",
        "\n",
        "res = requests.get(url, headers=headers)\n",
        "soup = BeautifulSoup(res.content, 'html.parser')\n",
        "soup"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Fpq03eZX-8jK",
        "outputId": "eaa58107-be49-44c2-cedd-39df0f85188a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<html lang=\"en\"><head><title>yelp.com</title><style>#cmsg{animation: A 1.5s;}@keyframes A{0%{opacity:0;}99%{opacity:0;}100%{opacity:1;}}</style></head><body style=\"margin:0\"><p id=\"cmsg\">Please enable JS and disable any ad blocker</p><script data-cfasync=\"false\">var dd={'rt':'c','cid':'AHrlqAAAAAMAJL-CcU9z8M8AJEyPvQ==','hsh':'3BD2468BAE4D73BEA0B5DE8314D745','t':'bv','qp':'find_desc%3Drestaurant%26find_loc%3DSan%2BFrancisco%252C%2BCA%26dd_referrer%3Dhttps%253A%252F%252Fwww.yelp.com%252F%26start%3D0','s':46977,'e':'8652525d23c07d2ec6b3f5100227be420e1fa18af63433602779153ee46759886438ef1a29bd07c6f6592aaf4bfa2782','host':'geo.captcha-delivery.com','cookie':'xLodcANXDSaFjf0tYU5fmMlZtuTWwtO0l2z1KXfuSoGgkNHX~7rr3z8~8pO4e3whIn9c24Buuxfpj1TcljoRqa0HjSxmeSrxpNfM627rdtJD2qUKexd3Mw9KFdBTlP6Q'}</script><script data-cfasync=\"false\" src=\"https://ct.captcha-delivery.com/c.js\"></script></body></html>"
            ]
          },
          "metadata": {},
          "execution_count": 4
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "\n",
        "# Replace with your actual Yelp API access token\n",
        "# ACCESS_TOKEN = \"xVy99wBbL08Z4lgyaB4_SRR04EPfRK2CKtCKAE_F47vZDaUtJnlVmxmYPxR2cfRIVuLkwQyjZInDWuFNSCNvXeF4iTcveMPugsJsDAeZRO6GCDla-t7ujHTuv77uXHYx\" # It's worked\n",
        "# ACCESS_TOKEN = \"zw1OcEcPG3dMWMlplhjQYfcHmo8QHpOaz6gEgoh4J2Ty9d4dSJV0_X5XGE8Bpv3BJUpPUsz9EfolmSDuzjQvSEgy4jKR-6_5FMxwnXI35MSU2zrOCvIUmUIr9AZrYnYx\"\n",
        "# ACCESS_TOKEN = \"U3LFmhPsqGDeHcf-2uVmQbNj03fI6Wp7gfQti7Ml0YctrPN3PhLwE3GEXT2htv3wJgz90ugTwq0FOap5hJ0rxUNUad81PyqAzmr3yuogEPLQ0jzoVrDEp3UeSkYzWXYx\"\n",
        "# ACCESS_TOKEN = \"k6Mwzc7QNv1vg1aUwAHQzBGe4AlDoU0KuslJlenhPaSX_ikbarWvvciZZt7ZB3Dd9UYE0Xn3_W6E0gr6SXgF2rWeS8LdFXDiaU_WM2TdBudJ-TxAMla9cnfaKIMDXXYx\"\n",
        "# ACCESS_TOKEN = \"YCiqpLcanOThy-y4Ubz2wRvtXez5dwrR_wQhQEADZFcI-rIhZoMu2GN_mGP_1YWkXQzPjoMbek1kTmFkCQ3LN7lZZqcprQV3yPV4BZXlGA8WPT-kYau2nY8C68Z7WXYx\"\n",
        "\n",
        "url = \"https://api.yelp.com/v3/graphql\"\n",
        "\n",
        "# GraphQL query (note the corrected syntax)\n",
        "query = \"\"\"\n",
        "{\n",
        "  search(term: \"restaurant\", location: \"san francisco\", limit: 50) {\n",
        "    total\n",
        "    business {\n",
        "      id\n",
        "    }\n",
        "  }\n",
        "}\n",
        "\"\"\"\n",
        "\n",
        "headers = {\n",
        "    \"Authorization\": f\"Bearer {ACCESS_TOKEN}\",\n",
        "    \"Content-Type\": \"application/graphql\"\n",
        "}\n",
        "\n",
        "response = requests.post(url, headers=headers, data=query)\n",
        "\n",
        "if response.status_code == 200:\n",
        "    data = response.json()\n",
        "    print(data)\n",
        "else:\n",
        "    print(f\"‚ùå Error {response.status_code}: {response.text}\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mTTuMxXO1ZxF",
        "outputId": "e1aa02e1-aa48-4115-a0b6-5512a6c582fa"
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'data': {'search': {'total': 5300, 'business': [{'id': 'QueFVMcMlT-6aZFv2M47mg'}, {'id': 'JAHzjTlZceEypYTnGziIjw'}, {'id': 'oV9fxdymGz6dMXcQdWaBfg'}, {'id': 'f-m7-hyFzkf0HSEeQ2s-9A'}, {'id': 'AH95PKaqfjhx79FfCEHCPQ'}, {'id': 'uUifmxOmL8ClHXyRVsmD3Q'}, {'id': '8dUaybEPHsZMgr1iKgqgMQ'}, {'id': 'lUUQi1b2rV3glIn4t2I1Iw'}, {'id': 'dY0Mc_PHH6nLENvWmVtwxQ'}, {'id': 'HHtpR0RslupSQ99GIIwW5A'}, {'id': 'keeuVo1Rgp2_5QN3t1REMQ'}, {'id': 'ZSzXw0NgJTyOzcHwKY5eMA'}, {'id': '0T1Kc6tUWaG3cPnyiD40Hw'}, {'id': 'mPsrI_pV0EQ4IfvbsLQ0Fg'}, {'id': 'BjlVE7tTp1UCEttgkM1Big'}, {'id': 'xpzXvAlHO0iBsfhPmYoktg'}, {'id': 'lz7EgaxedlmKRq9hrFkulA'}, {'id': 'pLBEfh2qKNa_9Z9V84XPRw'}, {'id': 'ttarnopezxmp2ROB1N2PaA'}, {'id': 'B09WOy0W83Od-Xw4xEXxog'}, {'id': 'kOYj2sGMZnyC1lU-_Uw5fA'}, {'id': 'o4V-TCAgB6_WQfRGvpBM3Q'}, {'id': 'THpTY0X_xwRzUL7wYD6dDg'}, {'id': '47QVdkSVaJcRPDwqhECwFA'}, {'id': '80k5mTDLv1Qc4PALrwKpoA'}, {'id': 'd5752S_RDb26HxoL6O06eQ'}, {'id': '_hOVIgjVRl_HzvLaa65KJg'}, {'id': 'zT4QiQc8ZUzqUnHZikob1g'}, {'id': 'J7_-faNq_Ag9qTOlDn81Pw'}, {'id': 'T6krutcCsZ317NS_EGbUZQ'}, {'id': 'K9XVDlPNhrrSVEJN7uWqJQ'}, {'id': 'zREcSPXvizDj-79lSw1MSQ'}, {'id': 's2R3RUJ09eBFvHVIUS4Shg'}, {'id': 'dTWMVRcMVC4zynbFQC7A2A'}, {'id': 'eYXwVR4mMAjzkJnm5wneHQ'}, {'id': '54FhPcut3uFDbpNCRWUOlQ'}, {'id': 'bwBIIsyJxDSPHXzDPNVZ0w'}, {'id': 'U7xNnqW_rXeo7G58JXr4uA'}, {'id': 'i09UMzccKgyLwGYKDVP28w'}, {'id': 'reXWH9Wo0ZTOuQsTMNOSxg'}, {'id': 'qfRZjzNqO3fRgEGqvGNY-A'}, {'id': 'MaaySW1ejpwQCkr9jn1glA'}, {'id': '3dCggyxAO0pAjG8n7usR4Q'}, {'id': 'wEpiKz--Z2b1q_FSCCNiow'}, {'id': 'D26qBnudK_1pdo6AlvMlvQ'}, {'id': 'x24zCTZhA-LiVgYMWs6Wqw'}, {'id': 'M8ZWmfOC9LJPV_7S06J5ZA'}, {'id': 'w8MW2jeXhOuNgNh8j3AEUA'}, {'id': 'PH3o6Te6QOA2VC2ZoYz6CQ'}, {'id': 'RjPKqHk_FiegJVCekbJ4Mw'}]}}}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "\n",
        "# SYGIC_KEY = \"EPrgMMQzpr9RMzaj25Tsw9QXrjtKbVMX2kY4NdWz\"\n",
        "# GOOGLE_KEY = \"AIzaSyD6F4xULR2I7GtEAH82L9vL6dAaEQAqnpQ\"\n",
        "# ACCESS_TOKEN = \"r1UIaTkCCqXMdMJC1Zu5KO5LYMU_k_e5eyttQlGbKcj7RYgued3tsikm4QnsX0GBRVVQ-1bIyxr6mSTD-Pl0MHrODWAxgakZF38cG-TVcnaiwc8idbB53rjQGwSDWnYx\"\n",
        "# ACCESS_TOKEN = \"HNOkS63Q7v5ski_gY7xF6lBzNBwNgaNBsGCB4-XhOSAK-hsLY8FqEyUt3Eguq9m1Q_QxNagyCdLZ5faHhL6in5ZYsvsVzlV-b0mAd6dYmwF9C4k5Jg-tGcUlDyk6Y3Yx\"\n",
        "# ACCESS_TOKEN = \"V5M8Oy8f8BAmG0WZ_FCt9opqcZSPhC5eossY8Onr2KSZuOwEDkWtGlSO3Ghda9BiJU4gT6d58vVzbzxMMJvmYDz4Z4aUrhIoC2ArI-DA94XXA9Qt2yUw1Ulv72qrX3Yx\"\n",
        "# ACCESS_TOKEN = \"2tbl7f9gBPKWCoTC2T8InIU64365FBnecXFOzDwyfElZAeLyoDmOZQu2Uc4k8Ttv0Lesc-FSQVLDBwv3D6JOy6Un1G4R3dDdqVdCNPMI7k-_emIZyGRdx7jxWG-PWXYx\"\n",
        "# ACCESS_TOKEN = \"6DqWyMUb2r5Hv0sXfYS5L1FlVqp83Fyn1XCEdbIRvchQVnl4ncuo_8opJcqxe6YjrLLmjZdp8r1HXv2nR8nnnxVt9DRh71uFq8MeUb_A0TxtbaJaj694yi-bZXMHW3Yx\"\n",
        "\n",
        "url = \"https://api.yelp.com/v3/businesses/search?location=san%20fransisco&term=restaurants&sort_by=best_match&limit=50\"\n",
        "\n",
        "headers = {\n",
        "    \"accept\": \"application/json\",\n",
        "    \"authorization\": f\"Bearer {ACCESS_TOKEN}\"\n",
        "}\n",
        "\n",
        "response = requests.get(url, headers=headers)\n",
        "\n",
        "response.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "CSrbCwYg3Ck3",
        "outputId": "94ac23cd-00df-4223-e924-d0f238487ce3"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'error': {'code': 'VALIDATION_ERROR',\n",
              "  'description': \"'Bearer ftPpQUCgfSA3yV98-uJn9g' does not match '^(?i)Bearer [A-Za-z0-9\\\\\\\\-\\\\\\\\_]{128}$'\",\n",
              "  'field': 'Authorization',\n",
              "  'instance': 'Bearer ftPpQUCgfSA3yV98-uJn9g'}}"
            ]
          },
          "metadata": {},
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "SYGIC_KEY = \"EPrgMMQzpr9RMzaj25Tsw9QXrjtKbVMX2kY4NdWz\"\n",
        "sygic_headers = {'x-api-key': SYGIC_KEY}\n",
        "\n",
        "AUSTIN_HOTEL = \"https://api.sygictravelapi.com/1.0/en/places/list?bounds=29.53522956294847,-98.4375,30.751277776257812,-97.03125&categories=sleeping&limit=500\"\n",
        "AUSTIN_EATING = \"https://api.sygictravelapi.com/1.0/en/places/list?bounds=29.53522956294847,-98.4375,30.751277776257812,-97.03125&categories=eating&limit=500\"\n",
        "AUSTIN_ATRACTION = \"https://api.sygictravelapi.com/1.0/en/places/list?bounds=29.53522956294847,-98.4375,30.751277776257812,-97.03125&categories=going_out|discovering|hiking|playing|sightseeing|doing_sports|relaxing&limit=500\"\n",
        "\n",
        "response = requests.get(AUSTIN_HOTEL, headers = sygic_headers)\n",
        "places = response.json()['data']['places']\n",
        "places"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 669
        },
        "id": "Bjq7PAmR6uiL",
        "outputId": "1a3e77e6-b2ae-4f96-fdd2-46cb2360f7e7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "error",
          "ename": "ConnectionError",
          "evalue": "HTTPSConnectionPool(host='api.sygictravelapi.com', port=443): Max retries exceeded with url: /1.0/en/places/list?bounds=29.53522956294847,-98.4375,30.751277776257812,-97.03125&categories=sleeping&limit=500 (Caused by NameResolutionError(\"<urllib3.connection.HTTPSConnection object at 0x7a7bc5853530>: Failed to resolve 'api.sygictravelapi.com' ([Errno -2] Name or service not known)\"))",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mgaierror\u001b[0m                                  Traceback (most recent call last)",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connection.py\u001b[0m in \u001b[0;36m_new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    197\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 198\u001b[0;31m             sock = connection.create_connection(\n\u001b[0m\u001b[1;32m    199\u001b[0m                 \u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_dns_host\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mport\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/util/connection.py\u001b[0m in \u001b[0;36mcreate_connection\u001b[0;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 60\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32min\u001b[0m \u001b[0msocket\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgetaddrinfo\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhost\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mport\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfamily\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msocket\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSOCK_STREAM\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     61\u001b[0m         \u001b[0maf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msocktype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mproto\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcanonname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msa\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.12/socket.py\u001b[0m in \u001b[0;36mgetaddrinfo\u001b[0;34m(host, port, family, type, proto, flags)\u001b[0m\n\u001b[1;32m    977\u001b[0m     \u001b[0maddrlist\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 978\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32min\u001b[0m \u001b[0m_socket\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgetaddrinfo\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhost\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mport\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfamily\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mproto\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    979\u001b[0m         \u001b[0maf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msocktype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mproto\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcanonname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msa\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mgaierror\u001b[0m: [Errno -2] Name or service not known",
            "\nThe above exception was the direct cause of the following exception:\n",
            "\u001b[0;31mNameResolutionError\u001b[0m                       Traceback (most recent call last)",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    786\u001b[0m             \u001b[0;31m# Make the request on the HTTPConnection object\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 787\u001b[0;31m             response = self._make_request(\n\u001b[0m\u001b[1;32m    788\u001b[0m                 \u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    487\u001b[0m                 \u001b[0mnew_e\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_wrap_proxy_error\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnew_e\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mproxy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscheme\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 488\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mnew_e\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    489\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    463\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 464\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_conn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    465\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m   1092\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_closed\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1093\u001b[0;31m             \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1094\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connection.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    752\u001b[0m             \u001b[0msock\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0msocket\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msocket\u001b[0m \u001b[0;34m|\u001b[0m \u001b[0mssl\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSLSocket\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 753\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msock\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msock\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_new_conn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    754\u001b[0m             \u001b[0mserver_hostname\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mstr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhost\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connection.py\u001b[0m in \u001b[0;36m_new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    204\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0msocket\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgaierror\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 205\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mNameResolutionError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhost\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    206\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mSocketTimeout\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mNameResolutionError\u001b[0m: <urllib3.connection.HTTPSConnection object at 0x7a7bc5853530>: Failed to resolve 'api.sygictravelapi.com' ([Errno -2] Name or service not known)",
            "\nThe above exception was the direct cause of the following exception:\n",
            "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    666\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 667\u001b[0;31m             resp = conn.urlopen(\n\u001b[0m\u001b[1;32m    668\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    840\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 841\u001b[0;31m             retries = retries.increment(\n\u001b[0m\u001b[1;32m    842\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnew_e\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_pool\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_stacktrace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_info\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/urllib3/util/retry.py\u001b[0m in \u001b[0;36mincrement\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    518\u001b[0m             \u001b[0mreason\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0merror\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mResponseError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcause\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 519\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mMaxRetryError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_pool\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreason\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mreason\u001b[0m  \u001b[0;31m# type: ignore[arg-type]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    520\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='api.sygictravelapi.com', port=443): Max retries exceeded with url: /1.0/en/places/list?bounds=29.53522956294847,-98.4375,30.751277776257812,-97.03125&categories=sleeping&limit=500 (Caused by NameResolutionError(\"<urllib3.connection.HTTPSConnection object at 0x7a7bc5853530>: Failed to resolve 'api.sygictravelapi.com' ([Errno -2] Name or service not known)\"))",
            "\nDuring handling of the above exception, another exception occurred:\n",
            "\u001b[0;31mConnectionError\u001b[0m                           Traceback (most recent call last)",
            "\u001b[0;32m/tmp/ipython-input-3179422865.py\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mAUSTIN_ATRACTION\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"https://api.sygictravelapi.com/1.0/en/places/list?bounds=29.53522956294847,-98.4375,30.751277776257812,-97.03125&categories=going_out|discovering|hiking|playing|sightseeing|doing_sports|relaxing&limit=500\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mAUSTIN_HOTEL\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msygic_headers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m \u001b[0mplaces\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'data'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'places'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0mplaces\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/requests/api.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     71\u001b[0m     \"\"\"\n\u001b[1;32m     72\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 73\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"get\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     57\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     58\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 59\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     60\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     61\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    587\u001b[0m         }\n\u001b[1;32m    588\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 589\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    590\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    591\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    701\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    702\u001b[0m         \u001b[0;31m# Send the request\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 703\u001b[0;31m         \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    704\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    705\u001b[0m         \u001b[0;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.12/dist-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    698\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mSSLError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    699\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 700\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    701\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    702\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mClosedPoolError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mConnectionError\u001b[0m: HTTPSConnectionPool(host='api.sygictravelapi.com', port=443): Max retries exceeded with url: /1.0/en/places/list?bounds=29.53522956294847,-98.4375,30.751277776257812,-97.03125&categories=sleeping&limit=500 (Caused by NameResolutionError(\"<urllib3.connection.HTTPSConnection object at 0x7a7bc5853530>: Failed to resolve 'api.sygictravelapi.com' ([Errno -2] Name or service not known)\"))"
          ]
        }
      ]
    }
  ]
}